design/one-pager-component-mutable-and-versioning.md

```diff
@@ -0,0 +1,499 @@
+# Component mutable and versioning mechanism
+
+* Owner: Lei Zhang (@resouer), Jianbo Sun (@wonderflow)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Terminology
+
+* Application: A holistic term for talking about all pieces that are rendered from an `ApplicationConfiguration`.
+* ComponentSchematic: A term in OAM spec v1alpha1, which is deprecated and replaced by `Component` in v1alpha2.
+
+## Background
+
+OAM v1alpha1 made a requirement that developer need to define another ComponentSchematic with
+different name for an update (i.e. ComponentSchematic is immutable). We've got lots of feedback from
+real practices in Alibaba as well as [AWS ECS for OAM](https://github.com/awslabs/amazon-ecs-for-open-application-model#upgrade-and-scale-oam-workloads-with-oam-ecs)
+that it's not a good experience. Today [Component is mutable in v1alpha2 spec](https://github.com/oam-dev/spec/pull/356),
+this proposal is going to make it into implementation.
+
+Since component is becoming mutable, a very common question is how will the component versioning mechanism work.
+
+One of the requirement is the versioning mechanism should allow users to specify a fixed version of component
+in ApplicationConfiguration, so that the component can be upgraded as before.
```

Comment:
```suggestion
One of the requirements that we should allow users to specify a fixed version of the component in an ApplicationConfiguration, so that the component will not be upgraded as soon as its definition changes.
```